lib/DDG/Spice/SentimentAnalysis/Document.pm

```diff
@@ -0,0 +1,31 @@
+package DDG::Spice::SentimentAnalysis::Document;
+# ABSTRACT: Returns result of sentiment analysis over a text or online content
+
+use DDG::Spice;
+
+primary_example_queries "sentiment analysis https://duck.co/ideas/idea/6767/sentiment-analysis-of-text-and-external-sources-in";
+description "Returns sentiment analysis of text or online content";
+name "Sentiment Analysis";
+code_url "https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/lib/DDG/Spice/SentimentAnalysis/Document.pm";
+topics "computing";
+category "language";
+attribution github  => ['https://github.com/gpestana', 'gpestana'],
+            twitter => ['https://twitter.com/gpestana', 'gpestana'];
+
+spice is_cached => 1;
+spice proxy_cache_valid => "200 1d";
+
+spice wrap_jsonp_callback => 1;
+spice to => 'http://gateway-a.watsonplatform.net/calls/url/URLGetTextSentiment?apikey={{ENV{DDG_SPICE_SENTIMENTANALYSIS_APIKEY}}}&url=$1&outputMode=json';
+
+triggers start => 'sentiment analysis', 'sentiment of', 'sa';
+
+handle remainder => sub {
+    #Extracts URL from remainder, if it exists
+    if ($_ =~ m!([--:\w?@%&+~#=]*\.[a-z]{2,4}\/{0,2})((?:[?&](?:\w+)=(?:\w+))+|[--:\w?@%&+~#=]+)?!) {
```

Comment:
I have a couple of questions regarding this. Optimally, we'll have to check if a certain URL is valid. The URL can range from domain (http://domain.com) to more complex URLs (e.g. domain.com/path/here#id or domain.com/path&article=10#something). As far as I got, the Data::Validate package does not offer any way to check the validity of all the different cases we need, as well as distinguishing between them.
With the approach used in the example you mentioned, we can check if the domain is valid and then pass the remainder as it is (eg. check if domain.com is valid and them pass domain.com/path#id to the Spice). Does this approach seem reasonable to you?
I think that to be 100% sure if the URL is valid + 'that it exists' is a bit of a pain. Probably we could loosen up the domain checker and adopt the solution you suggested. Or maybe I'm missing something?